harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -389,11 +389,15 @@ def _train_for_step(self, step_id: int, batches_per_step: int) -> workload.Respo
             if self.use_amp():
                 with apex.amp.scale_loss(loss, self.context.optimizer) as scaled_loss:
                     scaled_loss.backward()
+                    if self.hvd_config.use and communicate_and_update:
+                        # When using horovod, we need to finish communicating gradient
+                        # updates before they are unscaled which happens when.
```

Comment:
blocking: need to clarify the end of this comment a bit. Didn't quite finish saying when it actually happens.

Reply:
good catch, i just left off mid-sentence